overlord/snapstate/snapstate.go

```diff
@@ -1640,7 +1640,19 @@ func InstallPathMany(ctx context.Context, st *state.State, sideInfos []*snap.Sid
 		return nil, flagsByInstanceName[name], stateByInstanceName[name]
 	}
 
-	_, updateTss, err := doUpdate(ctx, st, names, updates, params, userID, flags, nil, deviceCtx, "")
+	var updateTss *UpdateTaskSets
+	if essential, nonEssential, ok := canSplitRefresh(deviceCtx, updates, flags); ok {
+		// if we're on classic with a kernel/gadget, split installs with essential
+		// snaps and apps so that the apps don't have to wait for a reboot
+		updateFunc := func(updates []minimalInstallInfo) ([]string, *UpdateTaskSets, error) {
+			// extra names are ignored so it's fine to passed all of them in each call
+			return doUpdate(ctx, st, names, updates, params, userID, flags, nil, deviceCtx, "")
+		}
+		_, updateTss, err = splitRefresh(essential, nonEssential, updateFunc)
+	} else {
+		_, updateTss, err = doUpdate(ctx, st, names, updates, params, userID, flags, nil, deviceCtx, "")
+	}
```

Comment:
should this be a wrapper function that can be used here and in `canSplitRefresh`?

Reply:
Hm it's a good question but I think in this case we'd be creating a wrapper with a very specific purpose (which is clearer if we leave it here I think). Also we'd be creating another function with a ton of parameters which is what I wanted to avoid with this. That's not a strong opinion though, maybe I'm wrong and it would make things clearer

Reply:
No you’re right, that’s the other side of the coin. Probably good as-is